proposals/handler.go

```diff
@@ -431,17 +456,28 @@ func (h *Handler) checkBallotDataIntegrity(b *types.Ballot) error {
 		if b.EpochData.Beacon == types.EmptyBeacon {
 			return errMissingBeacon
 		}
-		if len(b.ActiveSet) == 0 {
-			return errEmptyActiveSet
-		}
-		for i := 0; i < len(b.ActiveSet)-1; i++ {
-			if bytes.Compare(b.ActiveSet[i].Bytes(), b.ActiveSet[i+1].Bytes()) >= 0 {
-				return errActiveSetNotSorted
+		if len(b.ActiveSet) != 0 {
+			if err := h.handleSet(ctx, b.EpochData.ActiveSetHash, types.EpochActiveSet{
+				Epoch: b.Layer.GetEpoch(),
+				Set:   b.ActiveSet,
+			}); err != nil {
+				return err
+			}
+		} else {
+			if err := h.fetcher.GetActiveSet(ctx, b.EpochData.ActiveSetHash); err != nil {
```

Comment:
if this ref ballot is before the scheduled layer, this is an invalid ballot.
should only call h.fetcher.GetActiveSet() after the scheduled layer.
otherwise the node will be DOSing its peers on a bad ballot.

Reply:
why is it invalid? it for sure won't be rewarded, but otherwise such ballot is ok

Reply:
i think this is wrongheaded to try to solve this dos, when it is actually very simple work. and at the same time there are actually dos'able rpc's

Reply:
i added that condition

Reply:
yeah... there are many existing wrongs. 
here it seems a relative simple check to add.